x/finality/keeper/public_randomness.go

```diff
@@ -33,6 +34,30 @@ func (k Keeper) GetPubRandCommitForHeight(ctx context.Context, fpBtcPK *bbn.BIP3
 	return nil, types.ErrPubRandNotFound
 }
 
+// GetTimestampedPubRandCommitForHeight finds the public randomness commitment that includes the given
+// height for the given finality provider
+func (k Keeper) GetTimestampedPubRandCommitForHeight(ctx context.Context, fpBtcPK *bbn.BIP340PubKey, height uint64) (*types.PubRandCommit, error) {
```

Comment:
I prefer to have separate methods for different usages and the implementation is embedded. `HasTimestampedPubRand` is needed to return primitive values. Otherwise, `types.PubRandCommit` will cause circular dependency between btcstaking and finality modules